polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -129,6 +129,17 @@ void ValueCategory::store(mlir::OpBuilder &builder, mlir::Value toStore) const {
               builder.create<LLVM::NullOp>(nt.getLoc(), pt.getElementType());
       }
     }
+
+    if (Index) {
+      auto ElemTy =
+          pt.getElementType().cast<mlir::VectorType>().getElementType();
```

Comment:
How come you can cast  the element type to a vector type and not do a dyn_cast ? I guess is because Index is not null, wondering whether Is possible for `Index` to be non-null when the pointer element type is not a vector type?

Reply:
It is impossible, because the constructor with `Index` assert it:
```
ValueCategory::ValueCategory(mlir::Value Val, mlir::Value Index)
    : val{Val}, isReference{true}, Index{Index} {
  assert(
      ((val.getType().isa<MemRefType>() &&
        val.getType().cast<MemRefType>().getElementType().isa<VectorType>()) ||
       (val.getType().isa<LLVM::LLVMPointerType>() &&
        val.getType()
            .cast<LLVM::LLVMPointerType>()
            .getElementType()
            .isa<VectorType>())) &&
      "Expecting memref/pointer of vector");
  assert(Index.getType().isa<IntegerType>() && "Expecting integer index");
}
```